esrally/mechanic/provisioner.py

```diff
@@ -184,7 +186,17 @@ def _provisioner_variables(self):
         plugin_variables = {}
         mandatory_plugins = []
         for installer in self.plugin_installers:
-            mandatory_plugins.append(installer.plugin_name)
+            # For Elasticsearch < 6.3 more specific plugin names are required for mandatory plugin check
+            # Details in: https://github.com/elastic/elasticsearch/pull/28710
+            try:
+                if ((versions.major_version(self.distribution_version) == 6 and
+                     versions.minor_version(self.distribution_version) < 3) or
+                    (versions.major_version(self.distribution_version) < 6)):
+                    mandatory_plugins.append(installer.sub_plugin_name)
+                else:
+                    raise NameError
```

Comment:
I don't understand the need to raise `NameError` here.

Reply:
This is so that that if the version of Elasticsearch is >=6.3, the non-specific plugin name is applied (inside the exception code) i.e. force the execution inside the exception block.

Reply:
Hmm, but I mean it's not really an error condition if the version is >= 6.3? We could just set the plugin name with `mandatory_plugins.append(installer.plugin_name)` instead of raising an error which seems more straightforward to read to me.